ls1intum/Artemis
# [Feature] Modal for Import in Example Submission Board

### Checklist

### Motivation and Context



Follow-up PR for #3931 

This PR introduces new modal for importing example submission. It enables to instructor to see all student submissions are ready to import. That means the submission should be assessed by tutor/instructor to see it in the list. Only instructor can decide submission to be imported. Afterwards, this submission can be used as an example submission to train tutors. Submissions can be searched by participant name in the modal.

### Description

Added pageable query in `StudentParticipantRepository`. After retrieving participants for the exercise, it finds latest submission of student and makes pageable response for client side in `SubmissionService`. 

added `example-submission-import.component` and respective html template and spec file. It displays submissions in table. Also, uses `example-submission-import-paging.service` to retrieve the pageable submission content from server. and respective spec file added for paging service. 

added Import button in `example-submission.component`. It opens Import modal. To import the submission, it uses `exampleSubmissionService ` and sends the selected submission id and exercise id after closing the modal.

### Steps for Testing


1. Log in to Artemis with Instructor account
2. Navigate to Course Management
3. Find  or create Text/Modeling Exercise
4. Participate to this exercise with student account
5. Assess the student submission as tutor/instructor
6. Login as Instructor
7. Go to Example Submission board for the exercise
8. Click Import button
9. Confirm that you can see the submission in the table
10. Click Import button for that submission
11. Confirm that page is re-loaded and the new example submission is on the list
12. Then open the example submission
13. Confirm that submission is imported properly with its assessment

Plus: You can add more than one student submission for the exercise to see the submission list is retrieved properly when you open import modal for example submission.

### Review Progress





- Code Review
- Manual Tests

### Test Coverage



(only covered new implementation)
SubmissionResource.java 92%
SubmissionService.java 93%
example-submission-import-paging.service.ts 100%
example-submission-import.component.ts 96%

### Screenshots



Example Submission Board
<img src="https://user-images.githubusercontent.com/23650630/132963529-08a475de-9ea8-499c-9e83-9a23f08977c5.png" width="500"/>

Modal for Example Submission Import
<img src="https://user-images.githubusercontent.com/23650630/132963587-ea099903-15d9-4fe3-9e2e-39c2400ff917.png" width="500"/>
